src/components/AvatarWithImagePicker.js

```diff
@@ -116,112 +109,114 @@ const defaultProps = {
     originalFileName: '',
 };
 
-class AvatarWithImagePicker extends React.Component {
-    constructor(props) {
-        super(props);
-        this.animation = new SpinningIndicatorAnimation();
-        this.setError = this.setError.bind(this);
-        this.isValidSize = this.isValidSize.bind(this);
-        this.showAvatarCropModal = this.showAvatarCropModal.bind(this);
-        this.hideAvatarCropModal = this.hideAvatarCropModal.bind(this);
-        this.state = {
-            isMenuVisible: false,
-            validationError: null,
-            phraseParam: {},
-            isAvatarCropModalOpen: false,
-            imageName: '',
-            imageUri: '',
-            imageType: '',
-        };
-        this.anchorRef = React.createRef();
-    }
-
-    componentDidMount() {
-        if (!this.props.isUploading) {
-            return;
-        }
-
-        this.animation.start();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (!prevProps.isFocused && this.props.isFocused) {
-            this.setError(null, {});
-        }
-        if (!prevProps.isUploading && this.props.isUploading) {
-            this.animation.start();
-        } else if (prevProps.isUploading && !this.props.isUploading) {
-            this.animation.stop();
-        }
-    }
-
-    componentWillUnmount() {
-        this.animation.stop();
-    }
-
-    /**
+function AvatarWithImagePicker({
```

Comment:
We don't need `defaultProps` anymore, let's just set default values here as per https://expensify.slack.com/archives/C01GTK53T8Q/p1688785407639549

> destructuring props in function declarations is now documented as a best practice, and should be used instead of default props going forward. More info available at https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#destructuring

Reply:
Please take a look at this comment

Reply:
@eVoloshchak we need to keep the `defaultProps` as in some cases like this one it is not possible to remove the `defaultProps` because of the linter errors. I have consulted this and been asked to keep the `defaultProps` as they should be kept in JS files for know.